Modules/_ssl.c

```diff
@@ -2822,7 +2822,8 @@ _ssl_session_dup(SSL_SESSION *session) {
     }
     const_p = senc;
     newsession = d2i_SSL_SESSION(NULL, &const_p, slen);
-    if (session == NULL) {
+    if (newsession == NULL) {
+        PyErr_SetString(PyExc_ValueError, "d2i() failed.");
```

Comment:
Error messages in Python don't have a trailing dot: you should remove it from your added error message, and the two error messages above.